src/Network/Network/AzureFirewallPolicy/SetAzureFirewallPolicyCommand.cs

```diff
@@ -0,0 +1,159 @@
+    {
+
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+
+        [Alias("ResourceName")]
+        [Parameter(
+           Mandatory = true,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The resource name.", ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = SetByInputObjectParameterSet)]
+        [ResourceNameCompleter("Microsoft.Network/azureFirewalls", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.", ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The AzureFirewall Policy", ParameterSetName = SetByInputObjectParameterSet)]
+        public PSAzureFirewallPolicy InputObject { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(
+                    Mandatory = true,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "The resource Id.", ParameterSetName = SetByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public virtual string ResourceId { get; set; }
+
+        [Parameter(
+                    Mandatory = false,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "The operation mode for Threat Intelligence.")]
+        [ValidateSet(
+                    MNM.AzureFirewallThreatIntelMode.Alert,
+                    MNM.AzureFirewallThreatIntelMode.Deny,
+                    MNM.AzureFirewallThreatIntelMode.Off,
+                    IgnoreCase = false)]
+        public string ThreatIntelMode { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The base policy to inherit from")]
+        public string BasePolicy { get; set; }
+
+        [Parameter(
+                    Mandatory = true,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "location.", ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ParameterSetName = SetByResourceIdParameterSet)]
+        public virtual string Location { get; set; }
```

Comment:
Location should be in the `-InputObject` parameter set, and it should be NOT mandatory there

Reply:
More info: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/set-cmdlet-example.md#parameter-sets

Reply:
Fixed